contracts/wrapperSushi.sol

```diff
@@ -491,8 +491,10 @@ contract WrapAndUnWrapSushi is OwnableUpgradeable {
         }
     }
 
-    function conductUniswapT4T(address  [] memory theAddresses, uint amount) internal returns (uint256[] memory amounts1) {
+    function conductUniswapT4T(address  [] memory theAddresses, uint amount, uint256 userSlippageTolerance) internal returns (uint256[] memory amounts1) {
         uint256 deadline = 1000000000000000;
+        uint256 [] memory assetAmounts = getPriceFromSushiswap(theAddresses, amount);
+        require(amount > assetAmounts[1] + userSlippageTolerance * assetAmounts[1] / 100, "Insufficient Asset in Uniswap!");
```

Comment:
Move the min amount calculation from the `userSlippageTolerance` to a separate function all together please.